src/Symfony/Component/DependencyInjection/Argument/ServiceLocator.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Argument;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+final class ServiceLocator
```

Comment:
@chalasr and array is both an iterable and ArrayAccess like structure, the `IteratorArgument` allow to use `iterable` type hints and get either an array or an `IteratorArgument`. The idea behing implementing the ArrayAccess interface here is to allow the same for the ArrayAccess part : type-hinting `array|ArrayAccess` and receive either an array or a ServiceLocator.

Using ArrayAccess would allow the same code to be used inside the framework (by receiving a ServiceLocator) and outside the framework (by receiving an array).